python/dgl/backend/mxnet/tensor.py

```diff
@@ -368,28 +368,58 @@ def forward(self, lhs_data, rhs_data):
                             ctx=lhs_data.context, dtype=lhs_data.dtype)
         out_data_nd = zerocopy_to_dgl_ndarray_for_write(out_data)
         K.binary_op_reduce(
-            self.reducer, self.binary_op, self.graph, self.lhs, self.rhs,
+            self.reducer if self.reducer != 'mean' else 'sum',
+            self.binary_op, self.graph, self.lhs, self.rhs,
             lhs_data_nd, rhs_data_nd, out_data_nd, self.lhs_map[0],
             self.rhs_map[0], self.out_map[0])
+        # normalize if mean reducer
+        # note(zihao): this is a temporal hack and we should have better solution in the future.
+        if self.reducer == 'mean':
+            degs = nd.empty((out_data.shape[0],),
+                            ctx=out_data.context, dtype=out_data.dtype)
+            degs_nd = zerocopy_to_dgl_ndarray(degs)
+            if self.lhs != 1: # 1: target code of dst node
+                in_ones = nd.ones((lhs_data.shape[0],),
+                                  ctx=lhs_data.context, dtype=lhs_data.dtype)
+                in_ones_nd = zerocopy_to_dgl_ndarray(in_ones)
+                K.copy_reduce(
+                    'sum', self.graph, self.lhs, in_ones_nd, degs_nd, 
+                    self.lhs_map[0], self.lhs_map[0])
+            else: # self.rhs != 1
```

Comment:
I don't quite understand this if here. Shouldn't we always compute in-coming degrees?

Reply:
Yes we should always compute in-coming degrees.
When the input target is `src` or `edge`,  the output of this `copy_reduce` call is in-degrees of each node. But when the input target is `dst`, what's the expected behavior?
The `if` statement is to discriminate the case where lhs is on dst target (e.g. `v_op_e`)

Reply:
The real problem is `copy_reduce` does not accept input target as dst nodes.
```
dgl._ffi.base.DGLError: [03:39:04] /home/ec2-user/dgl/src/kernel/cpu/../binary_reduce_impl.h:104: Unsupported operation: op=use_lhs lhs=1 rhs=3
```
So I need to avoid this case.